osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -90,52 +82,56 @@ private void load()
             background.Colour = ColourProvider.Background5;
         }
 
-        [Resolved]
-        private Bindable<RulesetInfo> ruleset { get; set; }
-
         protected override void LoadComplete()
         {
             base.LoadComplete();
 
             header.Ruleset.BindTo(ruleset);
 
-            Country.BindValueChanged(_ =>
+            Country.BindValueChanged(country =>
             {
-                // if a country is requested, force performance scope.
-                if (Country.Value != null)
+                if (country.NewValue != null)
                     Scope.Value = RankingsScope.Performance;
-
-                Scheduler.AddOnce(loadNewContent);
             });
 
-            Scope.BindValueChanged(_ =>
+            Scope.BindValueChanged(scope =>
             {
-                // country filtering is only valid for performance scope.
-                if (Scope.Value != RankingsScope.Performance)
+                if (scope.NewValue != RankingsScope.Performance)
                     Country.Value = null;
 
-                Scheduler.AddOnce(loadNewContent);
+                Scheduler.AddOnce(selectDisplayToLoad);
             });
+        }
 
-            ruleset.BindValueChanged(_ =>
-            {
-                if (Scope.Value == RankingsScope.Spotlights)
-                    return;
+        private bool displayUpdateRequired = true;
 
-                Scheduler.AddOnce(loadNewContent);
-            });
+        protected override void PopIn()
+        {
+            base.PopIn();
+
+            // We don't want to create a new display on every call, only when exiting from fully closed state.
+            if (displayUpdateRequired)
+            {
+                header.Current.TriggerChange();
+                displayUpdateRequired = false;
+            }
+        }
```

Comment:
as far as I can see this was copied verbatim from `DashboardOverlay`. in that case maybe it should be added to the base in some way - but not necessarily in this PR as this one is bordering on too big already.

also, `displayUpdateRequired` looks like something `Cached` was designed for.